ethers-contract/ethers-contract-derive/src/abi_ty.rs

```diff
@@ -39,7 +39,14 @@ pub fn derive_tokenizeable_impl(input: &DeriveInput) -> proc_macro2::TokenStream
 
                 let assignments = fields.named.iter().map(|f| {
                     let name = f.ident.as_ref().expect("Named fields have names");
-                    quote_spanned! { f.span() => #name: #core_crate::abi::Tokenizable::from_token(iter.next().unwrap())? }
+                    let name_str = name.to_string();
+                    quote_spanned! { f.span() => #name: iter
+                        .next()
+                        .ok_or_else(|| #core_crate::abi::InvalidOutputType(::std::format!(
+                            "Struct field {:} doesn't have correspond token.", #name_str
+                        )))
+                        .and_then(|t| #core_crate::abi::Tokenizable::from_token(t))?
+                    }
```

Comment:
I think we should just use an `expect` here, because there's already a len check so this is guaranteed to succeed, hence the unwrap.

Reply:
Although I don't have problems with `expect`, there is also https://rust-lang.github.io/rust-clippy/master/#expect_used lint kind which some people might want to warn about it.

Reply:
this error can never happen though

Reply:
I know.Just want to pass the lint check for unwrap in my situation. If you want `expect`, then `expect` it is.